R/merge.R

```diff
@@ -81,9 +81,8 @@ merge.data.table = function(x, y, by = NULL, by.x = NULL, by.y = NULL, all = FAL
       yy = y[missingyidx]
       othercolsx = setdiff(nm_x, by)
       if (length(othercolsx)) {
-        tmp = rep.int(NA_integer_, length(missingyidx))
-        # TO DO: use set() here instead..
-        yy = cbind(yy, x[tmp, othercolsx, with = FALSE])
+        nx = make.unique(c(names(yy), othercolsx))
+        set(yy, NULL, tail(nx, length(othercolsx)), rep(list(NA), length(othercolsx)))
```

Comment:
What if missingindex is 1:nrow(y) and potentially copy is not happening (or may not happening in future due to some regression). Would be good to have a test that ensure that. Otherwise we could ended up updating y table.